frontend/src/pages/search.tsx

```diff
@@ -1,26 +1,71 @@
 import type { GetStaticProps, NextPage } from "next";
-import { useFeatureFlags } from "src/hooks/useFeatureFlags";
+import {
+  Opportunity,
+  SearchFetcher,
+  fetchSearchOpportunities,
+} from "../services/searchfetcher/SearchFetcher";
+import React, { useState } from "react";
 
+import { APISearchFetcher } from "../services/searchfetcher/APISearchFetcher";
+import { MockSearchFetcher } from "../services/searchfetcher/MockSearchFetcher";
+import PageNotFound from "./404";
 import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import { useFeatureFlags } from "src/hooks/useFeatureFlags";
 
-import PageNotFound from "./404";
+const useMockData = true;
+const searchFetcher: SearchFetcher = useMockData
+  ? new MockSearchFetcher()
+  : new APISearchFetcher();
 
-const Search: NextPage = () => {
+interface SearchProps {
+  initialOpportunities: Opportunity[];
+}
+
+const Search: NextPage<SearchProps> = ({ initialOpportunities = [] }) => {
   const { featureFlagsManager, mounted } = useFeatureFlags();
+  const [searchResults, setSearchResults] =
+    useState<Opportunity[]>(initialOpportunities);
+
+  const handleButtonClick = () => {
```

Comment:
We may want an event.preventDefault() here to prevent callback issues when the button is clicked.

Reply:
IIRC I think button inside a form will submit by default? But still added it for now